src/community/script/core/src/main/java/org/geoserver/script/rest/service/ScriptService.java

```diff
@@ -0,0 +1,214 @@
+                        }
+                    } : new Filter<org.geoserver.platform.resource.Resource>() {
+                        @Override
+                        public boolean accept(org.geoserver.platform.resource.Resource pathname) {
+                            return true;
+                        }
+                    };
+            for (org.geoserver.platform.resource.Resource f : Resources.list(dir, filter)) {
+                if (path.equals("apps")) {
+                    org.geoserver.platform.resource.Resource mainScript = scriptMgr.findAppMainScript(f);
+                    if (mainScript != null) {
+                        String name = mainScript.path().substring(
+                                f.parent().path().length() + 1).replace("\\", "/");
+                        scripts.add(new Script(name));
+                    }
+                } else if (path.equals("wps")) {
+                    if (f.getType() == Type.DIRECTORY) {
+                        String namespace = f.name();
+                        List<org.geoserver.platform.resource.Resource> files = f.list();
+                        for(org.geoserver.platform.resource.Resource file: files) {
+                            String name = namespace + ":" + file.name();
+                            scripts.add(new Script(name));
+                        }
+                    } else {
+                        String name = f.name();
+                        scripts.add(new Script(name));
+                    }
+                } else {
+                    String name = f.name();
+                    scripts.add(new Script(name));
+                }
+            }
+        } else {
+            // return empty array, perhaps we should return a 404?
+            // throw new RestletException(format("Could not find script dir %s",
+            // path),
+            // Status.CLIENT_ERROR_NOT_FOUND);
+        }
+
+        return scripts;
+    }
+
+    public void getScript(HttpServletRequest request, HttpServletResponse response) {
+        this.path = getPath(request);
+
+        org.geoserver.platform.resource.Resource script;
+        try {
+            if (path.contains(":")) {
+                path = path.replace(":","/");
+            }
+            script = scriptMgr.script(path);
+        } catch (IllegalStateException e) {
+            throw new RestException(format("Error looking up script %s", path),
+                    HttpStatus.INTERNAL_SERVER_ERROR, e);
+        }
+        if (!Resources.exists(script)) {
+            throw new RestException(format("Could not find script %s", path),
+                    HttpStatus.NOT_FOUND);
+        }
+
+        String fileContents;
+
+        try {
+            fileContents = FileUtils.readFileToString(script.file());
+            response.getWriter().write(fileContents);
+            response.setStatus(200);
+//            return new ResponseEntity<String>(fileContents, HttpStatus.OK);
```

Comment:
Again, why is this commented out? If it is not necessary, remove it.

Reply:
fixed.

Reply:
fixed.